pandas/plotting/_core.py

```diff
@@ -1746,69 +1746,66 @@ def _find_backend(backend: str):
+            if entry_point.name == backend:
+                module = entry_point.load()
+
+        if module is None:
+            # Fall back to unregistered, module name approach.
+            try:
+                module = importlib.import_module(backend)
+            except ImportError:
+                # We re-raise later on.
+                pass
+
+    if hasattr(module, "plot"):
+        # Validate that the interface is implemented when the option is set,
+        # rather than at plot time.
+        _backends[backend] = module
+        return module
 
     raise ValueError(
-        f"Could not find plotting backend '{backend}'. Ensure that you've installed "
-        f"the package providing the '{backend}' entrypoint, or that the package has a "
-        "top-level `.plot` method."
+        f"Could not find plotting backend '{backend}'. Ensure that you've "
+        f"installed the package providing the '{backend}' entrypoint, or that "
+        "the package has a top-level `.plot` method."
     )
 
 
-def _get_plot_backend(backend=None):
+def _get_plot_backend(backend: str | None = None):
     """
     Return the plotting backend to use (e.g. `pandas.plotting._matplotlib`).
 
-    The plotting system of pandas has been using matplotlib, but the idea here
-    is that it can also work with other third-party backends. In the future,
-    this function will return the backend from a pandas option, and all the
-    rest of the code in this file will use the backend specified there for the
-    plotting.
+    The plotting system of pandas uses matplotlib by default, but the idea here
+    is that it can also work with other third-party backends. This function
+    returns the module which provides a top-level `.plot` method that will
+    actually do the plotting. The backend is specified from a string, which
+    either comes from the keyword argument `backend`, or, if not specified, from
+    the option `pandas.options.plotting.backend`. All the rest of the code in
+    this file uses the backend specified there for the plotting.
 
     The backend is imported lazily, as matplotlib is a soft dependency, and
     pandas can be used without it being installed.
     """
     backend = backend or get_option("plotting.backend")
 
-    if backend == "matplotlib":
-        # Because matplotlib is an optional dependency and first-party backend,
-        # we need to attempt an import here to raise an ImportError if needed.
-        try:
-            import pandas.plotting._matplotlib as module
-        except ImportError:
-            raise ImportError(
-                "matplotlib is required for plotting when the "
-                'default backend "matplotlib" is selected.'
-            ) from None
-
-        _backends["matplotlib"] = module
-
     if backend in _backends:
         return _backends[backend]
 
-    module = _find_backend(backend)
-    _backends[backend] = module
-    return module
+    return _load_backend(backend)
```

Comment:
should return the backend module here and save it to `_backends` rather than mutating state in multiple places.